src/library/smartshape.lua

```diff
@@ -0,0 +1,84 @@
+--[[
+$module SmartShape
+]]
+local smartshape = {}
+
+
+--[[
+% smartshape_entrybased
+
+Creates an entry based SmartShape based on two input notes. If a type is not specified, creates a slur.
+
+@ start_note (FCNoteEntry) Starting note for SmartShape.
+@ end_note (FCNoteEntry) Ending note for SmartShape.
+@ type (SMARTSHAPE_TYPES or string)
+]]
+function smartshape.add_entry_based_smartshape(start_note, end_note, shape_type)
+    local smartshape = finale.FCSmartShape()
+    smartshape:SetEntryAttachedFlags(true)
+    -- Folks: Let me know if this is just too much... or not enough...
+    local types = {"slurauto" = finale.SMARTSHAPE_SLURAUTO, "slur_auto" = finale.SMARTSHAPE_SLURAUTO, "autoslur" = finale.SMARTSHAPE_SLURAUTO, "auto_slur" = finale.SMARTSHAPE_SLURAUTO, 
+"slur" = finale.SMARTSHAPE_SLURAUTO, "slurdown" = finale.SMARTSHAPE_SLURDOWN, "slur_down" = finale.SMARTSHAPE_SLURDOWN, "slurup" = finale.SMARTSHAPE_SLURUP, "slur_up" = finale.SMARTSHAPE_SLURUP,"dashed" = finale.SMARTSHAPE_DASHEDSLURAUTO, "dashedslur" = finale.SMARTSHAPE_DASHEDSLURAUTO, "dashed_slur" = finale.SMARTSHAPE_DASHEDSLURAUTO,"dashedslurdown" = finale.SMARTSHAPE_DASHEDSLURDOWN, "dashedslurup" = finale.SMARTSHAPE_DASHEDSLURDOWN, "dashedcurve" = finale.SMARTSHAPE_DASHCURVEAUTO, "dashed_curve" = finale.SMARTSHAPE_DASHCURVEAUTO,"curve" = finale.SMARTSHAPE_DASHCURVEAUTO,"dashedcurvedown", finale.SMARTSHAPE_DASHCURVEDOWN, "dashedcurveup"finale.SMARTSHAPE_DASHCURVEUP, "tabslide" = finale.SMARTSHAPE_TABSLIDE, "tab" = finale.SMARTSHAPE_TABSLIDE, "slide" = finale.SMARTSHAPE_TABSLIDE, "glissando" = finale.SMARTSHAPE_GLISSANDO, "gliss" = finale.SMARTSHAPE_GLISSANDO, "bendhat" = finale.SMARTSHAPE_BEND_HAT, "bend_hat" = finale.SMARTSHAPE_BEND_HAT,"hat" = finale.SMARTSHAPE_BEND_HAT, "bend" = finale.SMARTSHAPE_BEND_HAT, "bendcurve" = finale.SMARTSHAPE_BEND_CURVE, "bend_curve" = finale.SMARTSHAPE_BEND_CURVE}
+    --
+    if not shape_type then
```

Comment:
You can simplify this to the following:

```lua
shape_type = shape_type or "slur"
```

So basically, if `shape_type == nil`, then you assign it to "slur"